certbot-nginx/certbot_nginx/tls_sni_01.py

```diff
@@ -100,9 +100,13 @@ def _mod_config(self, ll_addrs):
             if line[0] == ['http']:
                 body = line[1]
                 found_bucket = False
+                posn = 0
                 for inner_line in body:
                     if inner_line[0] == bucket_directive[1]:
+                        if int(inner_line[1]) < int(bucket_directive[3]):
+                            body[posn] = [bucket_directive[1], bucket_directive[3]]
```

Comment:
This should be `body[posn] = bucket_directive`.

We do weird whitespace-hiding things on read, but it's actually there underneath so we need to add the whitespace for writing.

For example, check this out:

```
ipdb> inner_line
['server_names_hash_bucket_size', '32']
> body[posn] = bucket_directive
ipdb> body[posn]
['server_names_hash_bucket_size', '128']
ipdb> bucket_directive
['\n', 'server_names_hash_bucket_size', ' ', '128']
```